api/novelist.py

```diff
@@ -80,27 +103,30 @@ def values(cls, library):
             _db, ExternalIntegration.NOVELIST,
             ExternalIntegration.METADATA_GOAL, library=library
         )
+
         if not integration:
-            return (None, None)
+            return (None, None, None)
 
         profile = integration.username
         password = integration.password
-        return (profile, password)
+        authorized_identifier = integration.setting(cls.AUTHORIZED_IDENTIFIER).value
+        return (profile, password, authorized_identifier)
 
     @classmethod
     def is_configured(cls, library):
         if (cls.IS_CONFIGURED is None or
             library.id != cls._configuration_library_id
         ):
-            profile, password = cls.values(library)
-            cls.IS_CONFIGURED = bool(profile and password)
+            profile, password, authorized_identifier = cls.values(library)
+            cls.IS_CONFIGURED = bool(profile and password and authorized_identifier)
```

Comment:
Is NoveList always going to provide an authorized identifier? I kind of thought you might only have profile and password, with some other technique being used to populate the recommendation dataset.

Reply:
I wasn't sure about this but you can't call the API endpoint provided with just the profile and password set since the authorized identifier header is needed. Is it safe to say that not every library will call the API endpoint periodically and instead use the manual option? If that's the case then this check is not needed here.

Reply:
Changing over to two different integrations would also resolve this question.